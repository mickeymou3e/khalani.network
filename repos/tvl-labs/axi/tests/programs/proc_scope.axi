const a: Boolean;

let assumption := ~~a;
assume assumption {
    proc my_dn(x) {
        !dn(x)
    }
    let my_dn1 := |x| { !my_dn(x) };
    let scope_2 := |x| {
        let my_dn := || {
            !invalid(
            foo,
            bar,
            baz,
            )
        };
        # Test that my_dn in my_dn1 is referring to the one in the parent scope.
        !my_dn1(assumption)
    };
    let my_dn := || { !invalid() };
    !my_dn1(assumption)
}
!my_dn1(assumption)
