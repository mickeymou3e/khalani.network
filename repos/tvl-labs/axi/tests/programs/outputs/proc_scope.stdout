Result: Nothing
Result: Nothing
Result: Proved(if(not(not(a)), a))
